darray.h

```diff
@@ -21,19 +21,19 @@
 
 // Copy an element out of the array. Warning: not bounds checked.
 //
-// T rb_darray_get(rb_darray(T) ary, int32_t idx);
+// T rb_darray_get(rb_darray(T) ary, long idx);
```

Comment:
Any reason why you're using long instead of `size_t` ?

Reply:
That's a good point. Ruby's Array type uses `long` for it's capacity and sizes. I'm also not 100% sure about the overflow checking implications that @XrXr has implemented when changing from a signed integer to an unsigned integer.

Reply:
After discussions with @XrXr, we've decided to use `size_t` and use Ruby's builtin `malloc` with overflow checking.